project_profitability_fields/models/project_project.py

```diff
@@ -0,0 +1,35 @@
+from odoo import _, api, fields, models, exceptions
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+    
+    last_close_date = fields.Date()
+    estimated_cost = fields.Monetary()
+    estimated_income = fields.Monetary()
+    margin = fields.Float(
+        compute="_compute_margin",
+        inverse="_inverse_margin",
+        store=True,
+    )
+    @api.depends("estimated_income", "estimated_cost")
+    def _compute_margin(self):
+        for project in self:
+            if project.estimated_income:
+                project.margin = (
+                    project.estimated_cost
+                    and (project.estimated_income - project.estimated_cost)/project.estimated_income
+                    or 0.0
+                )
+            else:
+                project.margin = 0.0

```

Comment:
Why not this (original code replacing `estimated_cost` by `estimated_income` checking)?
```suggestion
            project.margin = (
                project.estimated_income
                and (project.estimated_income - project.estimated_cost)/project.estimated_income
                or 0.0
            )
```